deps/chakrashim/lib/chakra_inspector.js

```diff
@@ -0,0 +1,271 @@
+  };
+
+  var DebugManager = (function() {
+    var _internalHandle = Number.MAX_SAFE_INTEGER;
+
+    function GetNextInternalHandle() {
+      return --_internalHandle;
+    }
+
+    var _scripts = [];
+    var _scriptHandles = [];
+
+    var ScriptsManager = {
+      GetScripts: function() {
+        var scriptsArray = CallHostFunctionNoLog(chakraDebug.JsDiagGetScripts);
+        scriptsArray.forEach(function(element, index, array) {
+          if (!_scripts[element.scriptId]) {
+            var scriptHandle = GetNextInternalHandle();
+            _scriptHandles[scriptHandle] = scriptHandle;
+            _scripts[element.scriptId] = new V8Script(element, scriptHandle);
+          }
+        });
+        return _scripts;
+      },
+      GetScript: function(scriptId) {
+        if (!_scripts[scriptId]) {
+          this.GetScripts();
+        }
+        return _scripts[scriptId];
+      }
+    };
+
+    return {
+      ScriptsManager: ScriptsManager
+    };
+  })();
+
+
+  function V8Script(chakraScript, scriptHandle) {
+    this.chakraScript = chakraScript;
+    this.script = {};
+    this.script.handle = scriptHandle;
+    this.script.type = 'script';
+    this.script.name = ('fileName' in chakraScript) ?
+      chakraScript.fileName : chakraScript.scriptType;
+    this.script.id = chakraScript.scriptId;
+    this.script.lineOffset = 0;
+    this.script.columnOffset = 0;
+    this.script.lineCount = chakraScript.lineCount;
+    this.script.sourceStart = '';
+    this.script.sourceLength = chakraScript.sourceLength;
+    this.script.scriptType = 2;
+    this.script.compilationType = 0;
+    this.script.text = `${this.script.name} (lines: ${this.script.lineCount})`;
+  }
+
+  V8Script.prototype.GetId = function() {
+    return this.script.id;
+  };
+
+  V8Script.prototype.GetFileName = function() {
+    return this.chakraScript.fileName;
+  };
+
+  V8Script.prototype.GetScriptName = function() {
+    return this.name;
+  };
+
+  V8Script.prototype.GetHandle = function() {
+    return this.script.handle;
+  };
+
+  V8Script.prototype.GetProtocolObject = function() {
+    return this.script;
+  };
+
+  V8Script.prototype.IsSameFileName = function(fileName) {
+    if (fileName && this.GetFileName() &&
+      (this.GetFileName().toLowerCase() == fileName.toLowerCase())) {
+      return true;
+    }
+    return false;
+  };
+
+
+  function V8Breakpoint(type, target, scriptObject, line, column) {
+    this.type = type;
+    this.target = target;
+    this.scriptObject = scriptObject;
+    this.line = line;
+    this.column = column;
+  }
+
+  V8Breakpoint.prototype.GetTypeString = function() {
+    if (this.type == Debug.ScriptBreakPointType.ScriptId) {
+      return 'scriptId';
+    } else if (this.type == Debug.ScriptBreakPointType.ScriptName) {
+      return 'scriptName';
+    } else if (this.type == Debug.ScriptBreakPointType.ScriptRegExp) {
+      return 'scriptRegExp';
+    }
+  };
+
+  V8Breakpoint.prototype.Set = function() {
```

Comment:
It's based upon it.  The Debug.setBreakPoint function needs to be present in the Debug context in order for module.js to use it.  This is here to keep that functionality working, but I'm not planning to implement everything in JS like last time.